### ISSUE TITLE
Implement "assert_awaited" to CoroutineMock

### ISSUE THREAD

**Issue by peterdotran** [Martiusweb/asynctest#45]
```python
import asyncio
import asynctest.mock

async def async_func():
    ...
    some_async_action()
    ...

async def show_awaited():
    print('Did awaited')

some_async_action = asynctest.mock.CoroutineMock(side_effect=show_awaited)

loop = asyncio.get_event_loop()
loop.run_until_complete(async_func())
loop.close()

print(some_async_action.called)
some_async_action.assert_called()
```
some_async_action shows that is gets the coroutine some_async_action but it did not awaited. So there is an issue that would not be detected until runtime. I have a larger project and detecting that some functions that was properly awaited is going to help to find some issues of forgetting the awaited. Other cases I pass the result from the awaited function to another function and check that the call value was correct but that cannot be done in every case.

assert_awaited should assert that every call to the mock was awaited.

**Comment by Martiusweb** [comment]
That's a very good idea, I hope I'll find time to implement it.

**Comment by Kentzo** [comment]
I belive this task can be closed as being superseded by #64
